---
string[] job_ids
string[] types
string[] params
